enterprise/frontend/src/metabase-enterprise/audit_app/containers/AuditTableVisualization.jsx

```diff
@@ -104,6 +106,7 @@ export default class AuditTableVisualization extends React.Component {
       .map(({ name }) => _.findIndex(cols, col => col.name === name));
 
     if (rows.length === 0) {
+      onEmptyLoad();
```

Comment:
Invoking callback inside render is not a good practice because it potentially can lead to infinite re-renders, even though not in this case. It looks like you need a count of rows inside `ErrorOverview` to understand if the result was empty.
`QuestionLoadAndDisplay` has a prop `onLoad` which is used in `AuditTable` for the same goal of getting the number of returned rows. I'd use it similarly:
 add `onLoad` prop to the `AuditTable` and add a handler to pass it to `QuestionLoadAndDisplay`
```
  const handleLoad = results => {
    setLoadedCount(results[0].row_count);
    onLoad(results)
  }
```

Reply:
oh crap! didn't know that, thanks